products/compute/terraform.yaml

```diff
@@ -1140,6 +1140,56 @@ overrides: !ruby/object:Overrides::ResourceOverrides
         required: false
         default_from_api: true
         custom_flatten: 'templates/terraform/custom_flatten/name_from_self_link.erb'
+  RouterNat: !ruby/object:Overrides::Terraform::ResourceOverride
+    exclude_validator: true
+    id_format: "{{project}}/{{region}}/{{router}}/{{name}}"
+    mutex: router/{{region}}/{{router}}
+    examples:
+      - !ruby/object:Provider::Terraform::Examples
+        name: "router_nat_basic"
+        primary_resource_id: "nat"
+        skip_test: true
+        vars:
+          router_name: "my-router"
+          nat_name: "my-router-nat"
+          network_name: "my-network"
+          subnet_name: "my-subnetwork"
+      - !ruby/object:Provider::Terraform::Examples
+        name: "router_nat_manual_ips"
+        primary_resource_id: "nat_manual"
+        skip_test: true
+        vars:
+          router_name: "my-router"
+          nat_name: "my-router-nat"
+          network_name: "my-network"
+          subnet_name: "my-subnetwork"
+          address_name: "nat-manual-ip"
+    properties:
+      name: !ruby/object:Overrides::Terraform::PropertyOverride
+        validation: !ruby/object:Provider::Terraform::Validation
+          function: 'validateRFC1035Name(2, 63)'
+      natIps: !ruby/object:Overrides::Terraform::PropertyOverride
+        is_set: true
+      subnetwork: !ruby/object:Overrides::Terraform::PropertyOverride
+        is_set: true
+      subnetwork.sourceIpRangesToNat: !ruby/object:Overrides::Terraform::PropertyOverride
+        is_set: true
+      subnetwork.secondaryIpRangeNames: !ruby/object:Overrides::Terraform::PropertyOverride
+        is_set: true
+      region: !ruby/object:Overrides::Terraform::PropertyOverride
+        required: false
+        default_from_api: true
+        custom_flatten: 'templates/terraform/custom_flatten/name_from_self_link.erb'
+      minPortsPerVm: !ruby/object:Overrides::Terraform::PropertyOverride
+        default_from_api: true
```

Comment:
I think these (int values) should actually be Computed and send_empty_value = false - they can't be set to zero since all of them have validation from API that just denies values below a certain number (not sure if we should add config-level validation because they are arbitrary API numbers), so if you set the value, you can't unset it and we need to be able to read in the value from API if user unsets in config.

Reply:
after adding defaults, I think they no longer after to be computed